Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                        {
+                            //not in group- need to add it
+                            busFlow.AddActivity(stepActivity, tcActivsGroup);
+                        }
+
+                        //pull TC-Step parameters and add them to the Activity level
+                        List<string> stepParamsList = new List<string>();
+                        stepParamsList = GetTCParameterList(step.Description);
+                        //GetStepParameters(StripHTML(step.Expected), ref stepParamsList);
+                        foreach (string param in stepParamsList)
+                        {
+                            //get the param value
+                            string paramSelectedValue = string.Empty;
+                            bool? isflowControlParam = null;
+                            QC.QCTSTestParameter tcParameter = tc.Parameters.Where(x => x.Name.ToUpper() == param.ToUpper()).FirstOrDefault();
+
+                            //get the param value
+                            if (tcParameter != null && tcParameter.Value != null && tcParameter.Value != string.Empty)
+                                paramSelectedValue = tcParameter.Value;
+                            else
+                            {
+                                isflowControlParam = null;//empty value
+                                paramSelectedValue = "<Empty>";
+                            }
+
+                            //check if parameter is part of a link
+                            string linkedVariable = null;
+                            if (paramSelectedValue.StartsWith("#$#"))
+                            {
+                                string[] valueParts = paramSelectedValue.Split(new string[] { "#$#" }, StringSplitOptions.None);
+                                if (valueParts.Count() == 3)
+                                {
+                                    linkedVariable = valueParts[1];
+                                    paramSelectedValue = "$$_" + valueParts[2];//so it still will be considered as non-flow control
+
+                                    if (busVariables.Keys.Contains(linkedVariable) == false)
+                                    {
+                                        busVariables.Add(linkedVariable, valueParts[2]);
+                                    }
+                                }
+                            }
+
+                            //determine if the param is Flow Control Param or not based on it value and agreed sign "$$_"
+                            if (paramSelectedValue.StartsWith("$$_"))
+                            {
+                                isflowControlParam = false;
+                                if (paramSelectedValue.StartsWith("$$_"))
+                                    paramSelectedValue = paramSelectedValue.Substring(3);//get value without "$$_"
+                            }
+                            else if (paramSelectedValue != "<Empty>")
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)